### PR TITLE
Deploy Directus to Google Cloud Platform

### LINKED ISSUE #45: How to Run Directus to Google Cloud
### What is your idea?

In this article, I'll provide a comprehensive guide on how to run Directus on the Google Cloud Platform (GCP). I'll also cover setting up a GCP account, configuring Google Cloud SQL and Storage for Directus, and deploying Directus on a Google Cloud Compute Engine.

### What are the key takeaways from your post?

Readers will learn the following:

- How to set up a GCP account
- How to configure Google Cloud SQL
- How to configure Google Cloud Storage for Directus
- Setting up a  Compute Engine instance

### What type of post is this?

Self-Hosting Guide

### Country of residence

United Kingdom

### Terms & Conditions

- [X] You understand that submitting this form does not mean we accept your proposal. We will accept proposals at our discretion.
- [X] You understand that to be paid you will need to generate an invoice to us once the final draft of your work is completed.
- [X] You understand that you will grant us an unlimited license to post and promote your work as part of this program. You may repost, but using our copy as the canonical source.
- [X] You are aged 18 or over.

### PR DESCRIPTION
**Pull request by codejagaban**
This PR closes #45

### CODE REVIEW

**Comment by phazonoverload** [comment]
Hmm I doubt that any of these changes in the Dockerfile are actually necessary 🤔 
I feel like our standard directus/directus image can be used directly on GCP.

- pnpm would only be required to install additional packages, which isn't the case in this Dockerfile
- EXPOSE 8080 would only be required if the port would be changed via PORT env config, which doesn't seem to be the case
- chown shouldn't be necessary as we're already setting that ownership in the base image
- overriding the CMD would only be necessary if you want to bypass pm2 but I don't think that's advisable

**Comment by codejagaban** [comment]
> Hmm I doubt that any of these changes in the Dockerfile are actually necessary 🤔 I feel like our standard directus/directus image can be used directly on GCP.
> 
> * pnpm would only be required to install additional packages, which isn't the case in this Dockerfile
> * EXPOSE 8080 would only be required if the port would be changed via PORT env config, which doesn't seem to be the case
> * chown shouldn't be necessary as we're already setting that ownership in the base image
> * overriding the CMD would only be necessary if you want to bypass pm2, but I don't think that's advisable

Ah, you're correct! I just tried it, and it works perfectly fine
I used the previous configurations to run Directus locally with previous versions.

Updated!🙌🏼

File: deploy-directus-to-gcp/index.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    gcloud auth configure-docker us-central1-docker.pkg.dev
+    ```
+
+    The command above will configure Docker to authenticate with Google Artifact Registry.
+
+    ```
+    :::info Google Cloud Region
+    Wherever `us-central1` appears in this tutorial should be replaced with the `region` you selected when creating that resource
+    :::
+    ```
```
Comment (phazonoverload) [inline deploy-directus-to-gcp/index.md:112]:
````suggestion
1. Set up the Google Cloud CLI.

    Login to Google Cloud:

    ```bash
    gcloud auth login
    ```

    Select the project you created on Google Cloud via the `gcloud` CLI:

    ```bash
    gcloud config set project PROJECT_ID
    ```

    In the directory where the `Dockerfile` is located, configure Docker to authenticate with the Google Artifact Registry:

    ```bash
    gcloud auth configure-docker us-central1-docker.pkg.dev
    ```

    :::info Google Cloud Region
    Wherever `us-central1` appears in this tutorial should be replaced with the `region` you selected when creating that resource
    :::
````

File: deploy-directus-to-gcp/index.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+
+    ```bash
+    docker tag directus:10.9.2 us-central1-docker.pkg.dev/directus-project/directus-repo/directus:10.9.2
+    ```
+
+    Where:
+
+    - `directus-project` is the project ID you are working on
+    - `directus-repo` is the repository created
+    - `us-central1` is the region
```
Comment (phazonoverload) [inline deploy-directus-to-gcp/index.md:124]:
```suggestion
2. Tag the Docker Image.

    Tag the local docker image you built with the repository:

    ```bash
    docker tag directus:10.9.2 us-central1-docker.pkg.dev/directus-project/directus-repo/directus:10.9.2
    ```

    Where:
    - `directus-project` is the project ID you are working on
    - `directus-repo` is the repository created
    - `us-central1` is the region
```

File: deploy-directus-to-gcp/index.md [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    DB_USER: postgres
+    DB_PASSWORD: YOUR_DB_PASSWORD
+    DB_SSL__REJECT_UNAUTHORIZED: false
+    ```
+
+    ```
+    :::info `DB_HOST`
+    The `DB_HOST` is a combination of `/cloudsql/` and the connection name of your Cloud SQL database. This is a prefix indicating that the hostname is for a Cloud SQL instance.
+    :::
+    ```
```
Comment (phazonoverload) [inline deploy-directus-to-gcp/index.md:172]:
```suggestion
    :::info `DB_HOST`
    The `DB_HOST` is a combination of `/cloudsql/` and the connection name of your Cloud SQL database. This is a prefix indicating that the hostname is for a Cloud SQL instance.
    :::
```

File: deploy-directus-to-gcp/index.md [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+    STORAGE_GCS_BUCKET: YOUR_CLOUD_STORAGE_BUCKET_NAME
+    STORAGE_GCS_CREDENTIALS: YOUR_SERVICE_ACCOUNT_KEY_JSON
+
+    ```
+
+```
+:::info YOUR_SERVICE_ACCOUNT_KEY_JSON
+To create a new key for your service account, go to your Google Cloud console -> IAM & Admin -> Services Accounts -> Click on your project's Compute Engine default service account email and click on the Keys tab to create a new key.
+:::
+```
```
Comment (phazonoverload) [inline deploy-directus-to-gcp/index.md:200]:
```suggestion
:::info YOUR_SERVICE_ACCOUNT_KEY_JSON
To create a new key for your service account, go to your Google Cloud console -> IAM & Admin -> Services Accounts -> Click on your project's Compute Engine default service account email and click on the Keys tab to create a new key.
:::
```

**Review by phazonoverload — COMMENTED** [review]
Small changes

**Review by phazonoverload — APPROVED** [review]

**Comment by phazonoverload** [comment]
LGTM - you know the process from here. Thanks!

**Pull request closed without merging**